### ISSUE TITLE
Terminal panel keybinding doesn't work properly, 

### ISSUE THREAD

**Issue by non-descriptive** [zed-industries/zed#14740]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

After pressing Ctrl+\` zed opens new terminal, puts focus in command line and turns the keybinding into `terminal_panel::ToggleFocus` which on consecutive presses switch focus between terminal panel and main editor panel. In VsCode it would result in closing terminal panel, but there's no (documented) binding in zed to do that. There's a button in bottom right corner with exactly that functionality, but it suggests to use the same Ctrl+\`. 
Also there are several tooltips stating that Ctrl+~ (which is basically Ctrl+Shift+\` on most keyboards) should open new terminal tab in terminal panel, but it doesn't work at all and new terminal could be opened only via mouse clicks in open terminal panel. 

### Environment

Zed: v0.144.3 (Zed)
OS: Linux Wayland debian 12
Memory: 29.3 GiB
Architecture: x86_64

**Comment by JosephTLyons** [comment]
Hey @non-descriptive. Zed differentiates between the panel (the content of a dock) and a dock (the container that holds the content). Ctrl+\` does indeed run the `terminal_panel::ToggleFocus` command, which simply puts the terminal in focus. If the terminal dock is not open, it will open it, if it already open, it won't do anything but focus it. We have a separate commands for toggling the docks open and close. For instance, I keep my terminal in the bottom dock, so I use the `workspace: toggle bottom dock` command to toggle it. I have this bound to `cmd-b`.

https://github.com/user-attachments/assets/2176ad44-124e-4e1d-b7d8-4103203f9671

**Comment by non-descriptive** [comment]
Will try to rebound it and test it later. 
Shouldn't be button tooltip at the status line be different than? It hides the dock as you said, not toggles terminal visibility.

**Comment by gabriel-araujjo** [comment]
I have a similar issue.  I think it's related to this one, so I will explain it here.

When the editor has focus and I press `Crtl` + <code>`</code>, which in my keyboard layout (abnt2) requires the shift key being pressed, a literal grave accent appears on editor, and only after I press the combination again the focus changes to the integrated terminal.

Bellow there's a screencast of what's happening.

[Screencast from 2024-07-22 08-11-13.webm](https://github.com/user-attachments/assets/08d6638f-d542-42ab-9e7d-0c23fffd6ac1)

**Comment by non-descriptive** [comment]
Tested with
```json
[
  {
    "context": "Editor",
    "bindings": {
      "ctrl-`": "workspace::ToggleBottomDock"
    }
  },
  {
    "context": "Terminal",
    "bindings": {
      "ctrl-`": "workspace::ToggleBottomDock"
    }
  }
]
```

Looks like it works. Found funny how it mess up with you if you when you partially override it for certain context:
overrode `ctrl+b`/`ctrl+alt+b` to it for `Editor` context but not for `Terminal` - it toggles, focus on terminal and on second press - toggles sidebar/ai assistant instead and then change forcus to `Editor` context.
@gabriel-araujjo this might be your case - you didn't specified proper context. Try ones above. It prolly shouldn't matter since `qwerty` and `abnt2` use same set of bindings. Otherwise you prolly should open separate issue. 

@JosephTLyons is there some kind of void binding to override them to do nothing or should I make feature request for it? For example I don't want any AI assist to pop up, but I don't have anything else to bind to `ctrl+alt+b` yet. Empty strings yields validation error and reset the binding to default.

**Comment by JosephTLyons** [comment]
> Tested with
> 
> ```json
> [
>   {
>     "context": "Editor",
>     "bindings": {
>       "ctrl-`": "workspace::ToggleBottomDock"
>     }
>   },
>   {
>     "context": "Terminal",
>     "bindings": {
>       "ctrl-`": "workspace::ToggleBottomDock"
>     }
>   }
> ]
> ```
> 
> Looks like it works. Found funny how it mess up with you if you when you partially override it for certain context: overrode `ctrl+b`/`ctrl+alt+b` to it for `Editor` context but not for `Terminal` - it toggles, focus on terminal and on second press - toggles sidebar/ai assistant instead and then change forcus to `Editor` context. @gabriel-araujjo this might be your case - you didn't specified proper context. Try ones above. Layout prolly shouldn't matter since `qwerty` and `abnt2` use same set of bindings for tilda button. Otherwise you prolly should open separate issue.
> 
> @JosephTLyons is there some kind of void binding to override them to do nothing or should I make feature request for it? For example I don't want any AI assist to pop up, but I don't have anything else to bind to `ctrl+alt+b` yet. Empty strings yields validation error and reset the binding to default.

I think if you assign `null` to a binding within a context, the default action is no longer associated.

**Comment by gabriel-araujjo** [comment]
> ... @gabriel-araujjo this might be your case - you didn't specified proper context. Try ones above. Layout prolly shouldn't matter since qwerty and abnt2 use same set of bindings for tilda button. Otherwise you prolly should open separate issue.

@non-descriptive, thanks for the hint, but it doesn' t work for me. I'll open another issue.

**Comment by non-descriptive** [comment]
> I think if you assign null to a binding within a context, 

Nope. Looks like fallback to defaults. Anyway main rebinding is working, so I'm closing this issue.

@gabriel-araujjo looks like there's an issue(s) with switching layout locale as well and I've seen some related issues. Can you tag me when you find/open one.

**Comment by bnjmnt4n** [comment]
I feel like this behavior is quite unintuitive, because now I have to use two different shortcuts to open and close a single pane. It's also quite inconsistent that the behavior of the buttons can't be replicated by the shortcuts which are suggested in the tooltip, or by any other custom keybinding. Would it be possible to introduce new actions which can do this?
